src/api/spec/controllers/person_controller_spec.rb

```diff
@@ -20,4 +21,45 @@
       end
     end
   end
+
+  describe 'PUT #put_userinfo' do
+    let(:user) { create(:confirmed_user) }
+    let(:xml) {
+      <<-XML_DATA
+        <userinfo>
+          <realname>test name</realname>
+          <email>test@test.de</email>
+        </userinfo>
+      XML_DATA
+    }
+
+    context 'when in LDAP mode' do
+      before do
+        stub_const('CONFIG', CONFIG.merge({ 'ldap_mode' => :on }))
+        request.env["RAW_POST_DATA"] = xml
+      end
+
+      context 'as an admin' do
+        before do
+          login admin_user
+          put :put_userinfo, params: { login: user.login, format: :xml }
+        end
+
+        it 'is not allowed to perform this action' do
+          expect(response.header['X-Opensuse-Errorcode']).to eq('change_userinfo_no_permission')
```

Comment:
Always test what you expect to happen/not to happen, not only messages. In this case make sure the `User` doesn't change when you do this request.